src/style/utils/mix.js

```diff
@@ -0,0 +1,25 @@
+const decToHex = decimal => decimal.toString(16);
+const hexToDec = hex => parseInt(hex, 16);
+const isShortHex = hex => hex.length === 3;
+
+/**
+ * A Javascript implementation of the Sass `mix` mixin.
+ */
+export default (inputColorA, inputColorB, weight = 50) => {
+  let colorA = inputColorA.replace(/#/, ''),
+      colorB = inputColorB.replace(/#/, '');
+
+  colorA = isShortHex(colorA) ? colorA.repeat(2) : colorA;
+  colorB = isShortHex(colorB) ? colorB.repeat(2) : colorB;
```

Comment:
that should be redundant if you use ```/[^#]{2}/g``` regExp

Reply:
That regex will fail on hex codes with only three digits, however. I need to check the length of the string first (having before that stripped it of any #), repeat the string if necessary, and then find the pairs.

Reply:
my bad, I haven't looked closely

on the other hand it could be extracted to another method to not repeat code and increase readability